File: submissions/cs50x/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_coins(int cents);

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int sum = calculate_coins(cents);
    printf("%d\n", sum);
}

int calculate_coins(int cents)
{
    int quarters = 0;
    while (cents - 25 >= 0)
    {
        quarters++;
        cents = cents - 25;
    }

    int nickels = 0;
    while (cents - 10 >= 0)
    {
        nickels++;
        cents = cents - 10;
    }

    int pennies = 0;
    while (cents - 5 >= 0)
    {
        pennies++;
        cents = cents - 5;
    }

    int total = quarters + nickels + pennies + cents;
    return total;
}

File: submissions/cs50x/sentimental-cash/cash.py
total = 0


def main():
    while True:
        try:
            change = float(input("Change: "))
            if change < 0:
                raise
            break

        except:
            continue

    pennies(nickels(dimes(quarters(change))))
    print(total)


def quarters(n: float) -> float:
    global total
    while n >= 0.25:
        n -= 0.25
        total += 1
    return round(n, 2)


def dimes(n: float) -> float:
    global total
    while n >= 0.10:
        n -= 0.10
        total += 1
    return round(n, 2)


def nickels(n: float) -> float:
    global total
    while n >= 0.05:
        n -= 0.05
        total += 1
    return round(n, 2)


def pennies(n: float) -> float:
    global total
    while n >= 0.01:
        n -= 0.01
        total += 1
    return


main()
